pkg/util/log.go

```diff
@@ -0,0 +1,69 @@
+package util
+
+import (
+	"net"
+
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+)
+
+type Hook struct {
+	formatter logrus.Formatter
+	levels    []logrus.Level
+	fields    logrus.Fields
+	protocol  string
+	address   string
+}
+
+var (
+	fieldMap = logrus.FieldMap{
+		logrus.FieldKeyTime:  "@time",
+		logrus.FieldKeyLevel: "@level",
+		logrus.FieldKeyMsg:   "message",
+	}
+	fields = logrus.Fields{
+		"@key": uuid.New(), // This will be unique for every command being run
+	}
+)
+
+func NewHook(protocol, address string, levels []logrus.Level, formatter logrus.Formatter) *Hook {
+	logrus.New()
+	return &Hook{
+		formatter: formatter,
+		fields:    fields,
+		protocol:  protocol,
+		address:   address,
+		levels:    levels,
+	}
+}
+
+func (h *Hook) Fire(entry *logrus.Entry) error {
+	for k, v := range h.fields {
+		if _, ok := h.fields[k]; !ok {
+			h.fields[k] = v
+		}
+	}
+	for k, v := range h.fields {
+		entry.Data[k] = v
+	}
+	bytes, err := h.formatter.Format(entry)
+	if err != nil {
+		return err
+	}
+
+	w, err := net.Dial(h.protocol, h.address)
```

Comment:
I don't think its a good idea to do dial each time in Fire, should be enough to do it once in the NewHook method, simliar of how the syslog logrus hook.
https://github.com/sirupsen/logrus/blob/352781de903c9dc639752a3ac08148132746e180/hooks/syslog/syslog.go#L23-L25

Reply:
Ok, should be fixed now